### PR TITLE
Support MatrixBandPart and allow numLower to be a tensor

### PR DESCRIPTION
**Pull request by Linchenn**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7351)

### CODE REVIEW

**Review by mattsoulanille — COMMENTED** [review]
Looks good to me. Thank you!

Do we need a test in [`matrices_executor_test.ts`](https://github.com/tensorflow/tfjs/blob/master/tfjs-converter/src/operations/executors/matrices_executor.ts)?

File: tfjs-core/src/ops/linalg/band_part.ts [hunk 1]
```diff
@@ -72,48 +74,62 @@ import {zeros} from '../zeros';
...
+        () => `bandPart(): numLower (${numLower})` +
+            ` must not be greater than the number of rows (${M}).`);
+    $numLower =
+        convertToTensor(numLower < 0 ? M : numLower, 'numLower', 'bandPart') as
+        Scalar;
+  } else {
+    assert(
+        numLower.dtype === 'int32',
+        () => `bandPart(): numLower's dtype must be an int32.`);
+    // If numUpper is a Scalar, checking `numLower <= M` could hurt performance,
```
[the remark this answers is not in the document]
Comment (Linchenn) [reply tfjs-core/src/ops/linalg/band_part.ts]: Done.

**Comment by Linchenn** [comment]
> Looks good to me. Thank you!
> 
> Do we need a test in [`matrices_executor_test.ts`](https://github.com/tensorflow/tfjs/blob/master/tfjs-converter/src/operations/executors/matrices_executor.ts)?

Done. Thanks!

**Review by mattsoulanille — APPROVED** [review]
Thank you!

**Review by pyu10055 — APPROVED** [review]
LGTM, please also update the supported op file.

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7351#-:-NNrJq3621uxvTPK_Fkn:bpbk43f)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @Linchenn)

**Pull request merged**